deps/amqp_client/src/amqp_channel_sup.erl

```diff
@@ -59,22 +60,41 @@ start_writer(_Sup, direct, [ConnPid, Node, User, VHost, Collector, AmqpParams],
     end;
 start_writer(Sup, network, [Sock, FrameMax], ConnName, ChNumber, ChPid) ->
     GCThreshold = application:get_env(amqp_client, writer_gc_threshold, ?DEFAULT_GC_THRESHOLD),
-    supervisor2:start_child(
-      Sup,
-      {writer, {rabbit_writer, start_link,
+    StartMFA = {rabbit_writer, start_link,
                 [Sock, ChNumber, FrameMax, ?PROTOCOL, ChPid,
                  {ConnName, ChNumber}, false, GCThreshold]},
-       transient, ?WORKER_WAIT, worker, [rabbit_writer]}).
+    ChildSpec = #{id => writer, start => StartMFA,
+                  restart => transient, significant => true,
```

Comment:
Since the previous child spec was `transient` (not `intrinsic`) should we set `significant => false` here (or just remove the key)?

Reply:
Or, should it have been `intrinsic` all along?

Reply:
Yeah, I assumed `rabbit_writer` was significant enough. ;)

`rabbit_writer` performs internal buffering and flushes periodically. Therefore, if it crashes, I don't think we can just restart it and carry on - we lose the context/history of the data (to be) sent to the client, so I think we should terminate the channel at this point.

Reply:
Hmm how has it ever worked correctly then? 🤔 Makes sense to make it `significant`

Reply:
I think the way it works is that these processes virtually never crash and therefore their restart spec doesn't matter.  :)
I'm pretty sure the same is true for most other connection-related processes - they have some state and they can't recover it (it's only in the process), so restarting them is basically pointless - any crash of a connection related process should terminate the whole connection (perhaps for channel-related crashes, it's enough to restart the channel).